### PR TITLE
Clean up logs for Negotiate Authentication and Authorization

### LINKED ISSUE #21894: Cleanup logs in Authentication.Negotiate and Authorization
```
info: Microsoft.AspNetCore.Authorization.DefaultAuthorizationService[2]
      Authorization failed for These requirements were not met:
      DenyAnonymousAuthorizationRequirement:Requires an authenticated user.
```
Casing and spacing

```
info: Microsoft.AspNetCore.Authentication.Negotiate.NegotiateHandler[0]
      None
```
Uses `LogInformation` and lacks context

```
dbug: Microsoft.AspNetCore.Authentication.Negotiate.NegotiateHandler[6]
      Challenged 401 Negotiate
```
Should probably end with a `.`

cc @Tratcher

### PR DESCRIPTION
**Pull request by giuseppecampanelli**
Fixes #21894

### CODE REVIEW

**Comment by themilanfan** [comment]
fyi Ready for review now @Tratcher

File: src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs [hunk 1]
```diff
@@ -60,6 +61,10 @@ static NegotiateLoggingExtensions()
                 eventId: new EventId(10, "ClientError"),
                 logLevel: LogLevel.Debug,
                 formatString: "The users authentication request was invalid.");
+            _negotiateStateErrorType = LoggerMessage.Define<string>(
+                eventId: new EventId(11, "NegotiateStateErrorType"),
+                logLevel: LogLevel.Information,
+                formatString: "Negotiate State BlobErrorType of {blobErrorType}.");
```
Comment (Tratcher) [inline src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs:67]: This is an awkward error message. It doesn't seem meaningful the the reader.
```suggestion
                formatString: "Negotiate error code: {error}.");
```

File: src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs [hunk 2]
```diff
@@ -60,6 +61,10 @@ static NegotiateLoggingExtensions()
                 eventId: new EventId(10, "ClientError"),
                 logLevel: LogLevel.Debug,
                 formatString: "The users authentication request was invalid.");
+            _negotiateStateErrorType = LoggerMessage.Define<string>(
+                eventId: new EventId(11, "NegotiateStateErrorType"),
+                logLevel: LogLevel.Information,
```
Comment (Tratcher) [inline src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs:66]:
```suggestion
                logLevel: LogLevel.Debug,
```

File: src/Security/Authentication/Negotiate/src/NegotiateHandler.cs [hunk 3]
```diff
@@ -129,7 +129,7 @@ public async Task<bool> HandleRequestAsync()
                 _negotiateState ??= Options.StateFactory.CreateInstance();
 
                 var outgoing = _negotiateState.GetOutgoingBlob(token, out var errorType, out var exception);
-                Logger.LogInformation(errorType.ToString());
+                Logger.NegotiateStateErrorType(errorType.ToString());
                 if (errorType != BlobErrorType.None)
                 {
```
Comment (Tratcher) [inline src/Security/Authentication/Negotiate/src/NegotiateHandler.cs:134]:
```suggestion
                if (errorType != BlobErrorType.None)
                {
                    Logger.NegotiateStateErrorType(errorType.ToString());
```

File: src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs [hunk 4]
```diff
@@ -91,5 +96,8 @@ public static void CredentialError(this ILogger logger, Exception ex)
 
         public static void ClientError(this ILogger logger, Exception ex)
             => _clientError(logger, ex);
+
+        public static void NegotiateStateErrorType(this ILogger logger, string errorType)
```
Comment (Tratcher) [inline src/Security/Authentication/Negotiate/src/Internal/NegotiateLoggingExtensions.cs:100]:
```suggestion
        public static void NegotiateError(this ILogger logger, string error)
```

**Review by HaoK — APPROVED** [review]

**Comment by themilanfan** [comment]
@Tratcher thanks for the valuable feedback, it has been applied to the branch/pr

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**